### PR TITLE
(#96) Implemented Image.delete()

### PR DESCRIPTION
**Pull request by llorllale**
This PR:
* Solves #96 
* Leaves puzzle for the rest

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#101](https://github.com/amihaiemil/docker-java-api/issues/101) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) everybody who has role `REV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

File: src/main/java/com/amihaiemil/docker/RtImage.java [hunk 1]
```diff
@@ -47,14 +49,21 @@
      */
     private final URI baseUri;
 
+    /**
+     * The enclosing {@link Images} for this image.
+     */
+    private final Images parent;
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/RtImage.java]: @llorllale I don't think this is correct. "parent Images" does not refer to the ``Images`` instance which created this Image, but to those ``Images`` that are parents in terms of history, previous versions of this ``Image``, see the ``history`` operation. Makes sense?

Comment (llorllale) [reply src/main/java/com/amihaiemil/docker/RtImage.java]: @amihaiemil I see what you mean by "parent images"; I was just going for a "fluent" interface. I'll make it `void` then.

**Comment by llorllale** [comment]
@amihaiemil made `Image.delete()` return `void`, please check

**Comment by amihaiemil** [comment]
@llorllale nice, thanks :)

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
The job [#101](https://github.com/amihaiemil/docker-java-api/issues/101) is now out of scope

**Pull request merged**